torch/csrc/jit/script/module.h

```diff
@@ -85,7 +85,35 @@ struct Method {
     for (auto inp : member_inputs) {
       inputs.push_back(*inp);
     }
-    PropagateInputShapes(*retval, ArgumentSpec(with_grad, variable_tensor_list(std::move(inputs))));
+    PropagateInputShapes(
+      *retval,
+      ArgumentSpec(with_grad, variable_tensor_list(std::move(inputs))));
+    return retval;
+  }
+
+  std::shared_ptr<Graph> propagate_and_assign_input_and_output_shapes(std::vector<at::Tensor> inputs, std::vector<at::Tensor> outputs, bool with_grad=false, bool propagate=true) {
```

Comment:
We need to eventually revisit this, because it is not the right place to put this code, but we don't have complete story for how we provide size information anyway, so this can go in for now.